components/calendar/src/week_of.rs

```diff
@@ -37,34 +40,77 @@ impl From<&WeekDataV1> for WeekCalculator {
         Self {
             first_weekday: other.first_weekday,
             min_week_days: other.min_week_days,
+            weekend: None,
         }
     }
 }
 
 impl WeekCalculator {
-    icu_provider::gen_any_buffer_data_constructors!(
-        locale: include,
-        options: skip,
-        error: CalendarError,
-        /// Creates a new [`WeekCalculator`] from compiled locale data.
-        ///
-        /// ✨ *Enabled with the `compiled_data` Cargo feature.*
-        ///
-        /// [📚 Help choosing a constructor](icu_provider::constructors)
-    );
+    /// Creates a new [`WeekCalculator`] from compiled locale data.
+    ///
+    /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[cfg(feature = "compiled_data")]
+    pub fn try_new(locale: &DataLocale) -> Result<Self, CalendarError> {
+        Self::try_new_unstable(&crate::provider::Baked, locale)
+    }
+
+    #[doc = icu_provider::gen_any_buffer_unstable_docs!(ANY, Self::try_new_unstable)]
+    pub fn try_new_with_any_provider(
+        provider: &(impl AnyProvider + ?Sized),
+        locale: &DataLocale,
+    ) -> Result<Self, CalendarError> {
+        Self::try_new_unstable(&provider.as_downcasting(), locale).or_else(|e| {
```

Comment:
As @robertbastian mentioned above, you should make a new private constructor named something like `try_new_unstable_compat` which takes both V1 and V2 of the data struct, loads V2 if available, and otherwise loads V1 and converts it to V2 (filling in some plausible default value, probably the weekend set used in the root locale). There is an example of this in LocaleExpander and in https://github.com/unicode-org/icu4x/pull/2935

Reply:
wait what? this code already does that

Reply:
Yeah it might be nicer refactored into a helper function but yes this does appear to do the right thing